Proteomics/ProteolyticDigestion/Protease.cs

```diff
@@ -172,21 +170,84 @@ public CleavageSpecificity GetCleavageSpecificity(string proteinSequence, int st
         internal List<int> GetDigestionSiteIndices(string proteinSequence)
         {
             var indices = new List<int>();
-            for (int i = 0; i < proteinSequence.Length - 1; i++)
+            string proteinSubstring = proteinSequence;
```

Comment:
I'd like to see a lot of this code moved into DigestionMotif (I've added a few shells of methods that need completing there). Right now this class is kind of hard for me to follow. If we break this up into smaller methods in more appropriate places I think it'll be easier to understand